timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/LockWatchValueIntegrationTest.java

```diff
@@ -256,6 +261,52 @@ public void failedValidationCausesNoOpCacheToBeUsed() {
         });
     }
 
+    @Test
+    public void getRowsUsesCacheAsExpected() {
+        txnManager.runTaskThrowOnConflict(txn -> {
+            txn.put(TABLE_REF, ImmutableMap.of(CELL_1, DATA_1, CELL_2, DATA_2, CELL_3, DATA_3));
+            return null;
+        });
+
+        awaitUnlock();
+
+        Set<byte[]> rows = ImmutableSet.of("bar".getBytes(), "eggs".getBytes());
+        ColumnSelection columns = ColumnSelection.create(ImmutableSet.of("baz".getBytes(), "spam".getBytes()));
+
+        txnManager.runTaskThrowOnConflict(txn -> {
+            NavigableMap<byte[], RowResult<byte[]>> remoteRead = txn.getRows(TABLE_REF, rows, columns);
+            txn.delete(TABLE_REF, ImmutableSet.of(CELL_1));
+            assertHitValues(txn, ImmutableSet.of());
+            // we loaded all 4 values, but since we deleted one of them, it will not be in the digest
+            assertLoadedValues(
+                    txn,
+                    ImmutableMap.of(
+                            CellReference.of(TABLE_REF, CELL_2),
+                            CacheValue.of(DATA_2),
+                            CellReference.of(TABLE_REF, CELL_3),
+                            CacheValue.of(DATA_3),
+                            TABLE_CELL_4,
+                            CacheValue.empty()));
+            return remoteRead;
+        });
+
+        awaitUnlock();
+        // truncate the table to verify we are really using the cached values
+        truncateTable();
```

Comment:
This is a bit weird because we are actually doing a real read in the below transaction. Dropping or truncating is fine when we want to do 100% cached reads in a test, but I think this is a bit odd.

Reply:
I mean, it's just a different flavour of same jank. And to boot, the actual read is consistent with what was in the KVS, as that entry was deleted anyway

Reply:
That's true, but overwriting individual cells using that method would be better